src/main/java/org/jitsi/jigasi/SipCallTransformerMonitor.java

```diff
@@ -0,0 +1,228 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.jitsi.jigasi;
+
+import net.java.sip.communicator.service.protocol.media.*;
+import net.java.sip.communicator.util.*;
+import org.jitsi.impl.neomedia.*;
+import org.jitsi.impl.neomedia.rtcp.*;
+import org.jitsi.impl.neomedia.transform.*;
+import org.jitsi.service.neomedia.*;
+import org.jitsi.util.*;
+import org.jitsi.utils.*;
+import org.jitsi.utils.concurrent.*;
+
+import java.util.*;
+
+/**
+ * Monitors outgoing RTCP and passes only those which are relevant for that
+ * direction. Filters outgoing RTCP.BYEs for ssrc that we haven't seen. Also
+ * monitors last seen media or rtcp and if one is missing we
+ * send hole punch packets. It can happen that the xmpp call is dropped and
+ * the sip call participant is waiting then there will be no media ot RTCP to
+ * keep alive the stream from the other side.
+ *
+ * @author Damian Minkov
+ */
+public class SipCallTransformerMonitor
+    extends SinglePacketTransformerAdapter
+    implements TransformEngine
+{
+    /**
+     * The logger.
+     */
+    private final static Logger logger = Logger.getLogger(
+        SipCallTransformerMonitor.class);
+
+    /**
+     * The RTCP transformer.
+     */
+    private final RTCPTransformer rtcpTransformer = new RTCPTransformer();
+
+    /**
+     * List of seen incoming or outgoing ssrcs.
+     */
+    private Set<Long> seenSSRCs = new HashSet<>();
+
+    /**
+     * The executor which periodically calls {@link KeepAliveIncomingMedia}.
+     */
+    private static final RecurringRunnableExecutor EXECUTOR
+        = new RecurringRunnableExecutor(KeepAliveIncomingMedia.class.getName());
+
+    /**
+     * The runner that checks media.
+     */
+    private KeepAliveIncomingMedia recurringMediaChecker
+        = new KeepAliveIncomingMedia(25000);
+
+    private long lastOutgoingActivity;
+
+    /**
+     * The peer handler for which we are adding this transformer.
+     */
+    private final CallPeerMediaHandler handler;
+
+    /**
+     * The stream for this transformer.
+     */
+    private final MediaStream stream;
+
+    /**
+     * Initializes a new {@link SsrcRewriter} instance.
+     */
+    public SipCallTransformerMonitor(
+        CallPeerMediaHandler handler, MediaStream stream)
+    {
+        super(RTPPacketPredicate.INSTANCE);
+
+        this.handler = handler;
+        this.stream = stream;
+
+        EXECUTOR.registerRecurringRunnable(recurringMediaChecker);
+    }
+
+    /**
+     * Disposes, clears the executor and its task.
+     */
+    void dispose()
+    {
+        EXECUTOR.deRegisterRecurringRunnable(recurringMediaChecker);
+        EXECUTOR.close();
```

Comment:
Do we really want to close the static executor when we dispose of a single instance?

Reply:
Oh :) will fix it.